beacon-chain/core/electra/withdrawals.go

```diff
@@ -148,7 +161,8 @@ func ProcessWithdrawalRequests(ctx context.Context, st state.BeaconState, wrs []
 			// Only exit validator if it has no pending withdrawals in the queue
 			if pendingBalanceToWithdraw == 0 {
 				var err error
-				st, err = validators.InitiateValidatorExit(ctx, st, vIdx, validators.ExitInformation(st))
+				// exitInfo is updated within InitiateValidatorExit
+				st, err = validators.InitiateValidatorExit(ctx, st, vIdx, exitInfo)
```

Comment:
if we don't like doing the exit info thing here maybe we just pass in a function for initiating the exit like
```
// Usage in withdrawal processing
  var computeExit ExitEpochComputer
  if st.Version() < version.Electra {
      exitInfo := validators.ExitInformation(st)
      computeExit = computeExitEpochPreElectra(exitInfo)
  } else {
      computeExit = computeExitEpochElectra()
  }

  for _, wr := range withdrawalRequests {
      // ...
      st, err = validators.InitiateValidatorExit(ctx, st, vIdx, computeExit)
      // ...
  }
  ```

Reply:
reduces the fork checks

Reply:
yeah, an enclosure may be good, but it makes the code on process operations a little uglier cause the exit info is there used on three different callees.